pyiron_base/project/generic.py

```diff
@@ -581,14 +581,41 @@ def iter_jobs(
             case, you may seriously wish to consider setting `convert_to_object=False` and access only the HDF5/JobCore
             representation of the jobs instead.
         """
-        job_id_lst = self.job_table(recursive=recursive, **kwargs)["id"]
-        if progress:
-            job_id_lst = tqdm(job_id_lst)
-        for job_id in job_id_lst:
-            if path is not None:
-                yield self.inspect(job_id)[path]
-            else:  # Backwards compatibility - in future the option convert_to_object should be removed
-                yield self.load(job_id, convert_to_object=convert_to_object)
+        if not isinstance(self.db, FileTable):
+            job_id_lst = self.job_table(recursive=recursive, **kwargs)["id"]
+            if progress:
+                job_id_lst = tqdm(job_id_lst)
+            for job_id in job_id_lst:
+                if path is not None:
+                    yield self.inspect(job_id)[path]
+                else:  # Backwards compatibility - in future the option convert_to_object should be removed
+                    yield self.load(job_id, convert_to_object=convert_to_object)
+        else:
+            db_entry_lst = [row[1].to_dict() for row in self.job_table().iterrows()]
+            table_columns = [
+                "job",
+                "subjob",
+                "projectpath",
+                "project",
+                "status",
+                "hamilton",
+                "hamversion",
+            ]
```

Comment:
I notice this is only a subset of the columns mentioned in [`pyiron_base.database.filetable.table_columns`](https://github.com/pyiron/pyiron_base/blob/59d0852ddba64490535de203fa3cf99a9fba13b6/pyiron_base/database/filetable.py#L30) and in the [full column list for the job table](https://github.com/pyiron/pyiron_base/blob/59d0852ddba64490535de203fa3cf99a9fba13b6/pyiron_base/database/generic.py#L178). Will this truncation possibly get us into trouble? If the truncation is intentional it might be helpful to future maintainers to add a short comment on the intent of the truncation.

Reply:
We do not want to use the `id`, `masterid` or `parentid` as these can lead to inconsistencies when using pyiron without a database. In `pyiron_base.jobs.job.path` we define the `from_db_entry()` function: 
```
    @classmethod
    def from_db_entry(cls, db_entry):
        """
        Load a job path from a database entry.

        Args:
            db_entry (dict): database entry {"job":, "subjob":, "projectpath":, "project":, "hamilton":, "hamversion":,
                                            "status":} and optional entries are {"id":, "masterid":, "parentid":}
        """
        hdf5_file = db_entry["subjob"].split("/")[1] + ".h5"
        job_path = db_entry["projectpath"]
        if job_path is None:
            job_path = ""
        job_path += db_entry["project"] + hdf5_file + db_entry["subjob"]
        job = cls(job_path=job_path)

        if "hamilton" in db_entry.keys():
            job.__name__ = db_entry["hamilton"]
        if "hamversion" in db_entry.keys():
            job.__version__ = db_entry["hamversion"]

        if "id" in db_entry.keys():
            job._job_id = db_entry["id"]
        if "status" in db_entry.keys():
            job._status = db_entry["status"]
        if "masterid" in db_entry.keys():
            job._master_id = db_entry["masterid"]
        if "parentid" in db_entry.keys():
            job._parent_id = db_entry["parentid"]
        return job
```
So I selected all columns which can be used by this function and are not related to the job id.

Reply:
That sounds perfectly reasonable to me! In the code the list just appears out of thin air, and I really think it would help future developers to have a truncated version of that comment appear next to the list defined here.

Reply:
I added a short comment.